test/parallel/test-copy.js

```diff
@@ -0,0 +1,197 @@
+'use strict';
+
+const common = require('../common');
+if (!common.hasCrypto) { common.skip('missing crypto'); }
+
+const assert = require('assert');
+const { randomUUID } = require('crypto');
+const fs = require('fs');
+const {
+  copy,
+  copySync,
+  lstatSync,
+  mkdirSync,
+  readdirSync,
+  symlinkSync,
+} = fs;
+const net = require('net');
+const { dirname, join } = require('path');
+
+const isWindows = process.platform === 'win32';
+const tmpdir = require('../common/tmpdir');
+tmpdir.refresh();
+
+let dirc = 0;
+function nextdir() {
+  return join(tmpdir.path, `copy_${++dirc}`);
+}
+
+// Async version of copy.
+
+// It copies a nested folder structure with files, folders, symlinks.
+{
+  const src = dirname(require.resolve('../fixtures/copy/kitchen-sink'));
+  const dest = nextdir();
+  copy(src, dest, common.mustCall((err) => {
+    assert.strictEqual(err, null);
+    assertDirEquivalent(src, dest);
+  }));
+}
+
+// It throws error if existing symlink in dest is in subdirectory src.
+// TODO(bcoe): this behavior seemed strange to me, ask about it in review.
```

Comment:
@manidlou if I copy the `kitchen-sink` folder, which contains symlinks, twice to a `dest` folder, on the second copy I get `ERR_FS_COPY_TO_SUBDIRECTORY`.

This seemed odd to me because the `dest` folder is not a subdirectory of `src`.

Reply:
I also noticed that this behavior only errors on *inux and seemingly not Windows.

Reply:
This behavior I believe ended up being a bug related to relative symlinks, I corrected by adding this logic:

```js
      if (!isAbsolute(resolvedDest)) {
        resolvedDest = resolve(dirname(dest), resolvedDest);
      }
```

Reply:
@bcoe I see, that makes sense. Thanks for the fix!